src/data_vis.ts

```diff
@@ -124,51 +116,43 @@ class DataVisModel {
     filteredTSTable: KnockoutObservableArray<any>;
 
     constructor() {
-        this.currentPlot.subscribe(function() {
-            if (this.currentPlot() == "Impact") {
-                this.renderImpact();
-            } else if (this.currentPlot() == "Time series") {
-                this.renderTimeSeries();
-            }
-        }, this)
-        this.compare.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.disaggregateBy.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.cumulativePlot.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.maxBars.subscribe(function() {
-            this.onUIChange(true, false); // this has to be false to prevent infinite regression
-        }, this)
-        this.yearFilter().selectedLow.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.yearFilter().selectedHigh.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.activityFilter().selectedOptions.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.countryFilter().selectedOptions.subscribe(function() {;
-            this.onUIChange(true, true);
-        }, this)
-        this.diseaseFilter().selectedOptions.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.vaccineFilter().selectedOptions.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-        this.touchstoneFilter().selectedOptions.subscribe(function() {
-            this.onUIChange(true, true);
-        }, this)
-
-        this.compareNames.subscribe(function() {
-            this.maxPlotOptions(createRangeArray(1, this.compareNames().length));
-            this.maxBars(this.compareNames().length);
-        }, this)
+
+        this.canvas = document.getElementById('myChart');
```

Comment:
I noticed these canvas/context objects were being recreated multiple times, but I don't think they need to be? So I moved them into the constructor where they will be initialised just once.